src/components/AppBar.jsx

```diff
@@ -0,0 +1,173 @@
+import React, { useState, useEffect } from "react";
+import {
+  AppBar,
+  Toolbar,
+  Typography,
+  IconButton,
+  Button,
+  Avatar,
+  Menu,
+  MenuItem,
+  TextField,
+  Grid,
+} from "@mui/material";
+import SearchIcon from "@mui/icons-material/Search";
+import CancelIcon from "@mui/icons-material/Cancel";
+import { useParams, useNavigate } from "react-router-dom";
+import users from "../users";
+
+function AppBarComponent() {
+  const [isSearchOpen, setSearchOpen] = useState(false);
+  const [isDropdownOpen, setDropdownOpen] = useState(false);
+  const [avatarAnchorEl, setAvatarAnchorEl] = useState(null);
+  const [searchInput, setSearchInput] = useState("");
+  const navigate = useNavigate();
+
+  const handleSearchClick = () => {
+    setSearchOpen(!isSearchOpen);
+  };
+
+  const handleDropdownClick = (event) => {
+    setAvatarAnchorEl(event.currentTarget);
+    setDropdownOpen(true);
+  };
+
+  const closeDropdown = (event) => {
```

Comment:
Very bad practice.
The one that should handle the scenarios is the component, or it's father.
not a freaking closeDropDown event!